flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -184,7 +183,9 @@ public YarnResourceManager(
 		this.defaultCpus = flinkConfig.getInteger(YarnConfigOptions.VCORES, numberOfTaskSlots);
 		this.resource = Resource.newInstance(defaultTaskManagerMemoryMB, defaultCpus);
 
-		this.slotsPerWorker = updateTaskManagerConfigAndCreateWorkerSlotProfiles(flinkConfig, defaultTaskManagerMemoryMB, numberOfTaskSlots);
+		this.defaultTaskManagerParameters = ContaineredTaskManagerParameters.create(flinkConfig, defaultTaskManagerMemoryMB, numberOfTaskSlots);
```

Comment:
Yes, the `createTaskExecutorLaunchContext ` recreates a `ContaineredTaskManagerParameters`, but with the actual container memory allocated by Yarn, which could be slightly larger than requested. Reusing the same `ContaineredTaskManagerParameters` instance prevents task managers from making use of the extra allocated memory, which could be a user aware behavior change compare to the current behavior. I think maybe we should not introduce such behavior change in this version, since we are going to change resource configuration and management anyway in the next version.